src/main/java/org/example/Phone.java

```diff
@@ -1,37 +1,96 @@
 package org.example;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-interface Phone {
+public abstract class Phone {
+    public String color;
+    protected String material;
+    protected String IMEI;
 
-    public List<String> addContact(Integer contactCriteriaNumber, String phoneNumber, String firstName, String lastName);
+    public Phone(String IMEI, String material, String color) {
+        this.IMEI = IMEI;
+        this.material = material;
+        this.color = color;
 
-    void call(String phoneNumber);
-    String[] sendMessage(String s, String[] strings);
+    }
 
-    String[] getFirstMessage(String phoneNumber);
-    String [] getSecondMessage(String phoneNumber);
+    Integer BATTERY_LIFE = 24;
+    public List<String> contactList = new ArrayList<>();
+    List<String> callHistory = new ArrayList<>();
+    List<String> messageHistoryForSpecificPhoneNumber = new ArrayList<>();
+    public Map<String, List<String>> messageList = new HashMap<>();
 
-    void getContact(int i);
+    public void addContact(Contact contact) {
+        contactList.add(contact.toString());
+    }
 
+    public String getFirstContact() {
+        return contactList.get(0);
+    }
 
+    public String getLastContact() {
+        return contactList.get(contactList.size() - 1);
+    }
 
+    public void seeAllExistingContacts() {
+    }
 
-//    public int call(String phoneNumber);
-//
-//    public void getSecondMessage(String phoneNumber);
-//
-//    public void getFirstMessage(String phoneNumber);
-//
-//    public int sendMessage(String phoneNumber, String[] messageContent);
-//
-//    public void getLastContact();
-//
-//    public void getFirstContact();
-//
-//    public void SeeExistingContacts();
-//
+    public void sendTextMessage(String phoneNumber, String textMessage) {
+        if (textMessage.length() < 500) {
+            messageHistoryForSpecificPhoneNumber.add(textMessage);
+            BATTERY_LIFE = BATTERY_LIFE - 1;
+        }
+        messageList.put(phoneNumber, messageHistoryForSpecificPhoneNumber);
+    }
 
-//    public void setContactList();
-}
+    public String getFirstMessage(String phoneNumber) {
+        for (Map.Entry<String, List<String>> messageList : messageList.entrySet()) {
+            if (messageList.getKey() == phoneNumber) {
+                return messageHistoryForSpecificPhoneNumber.get(0);
```

Comment:
You could create a message object with Contact and content fields. Then add a List of Message inside the phone and always query that list